### ISSUE TITLE
Helix crashes while editing yaml

### ISSUE THREAD

**Issue by kharf** [helix-editor/helix#7894]
### Summary

I have been using Helix 23.05 for quite some time and been editing the same yamls over and over again, but today Helix just crashes as soon as I want to edit some of the yamls (not all).
I don't have the yaml-language-server installed. When I installed it today, I could actually edit the files, but only sometimes.

I could reproduce it on of my public repositories: https://github.com/kharf/flux-k8s-stack as described below.

An example where helix doesn't crash is the `catalog/keda/releases.yaml`

### Reproduction Steps

I tried this:

1. `cd flux-k8s-stack`
2. `hx .`
3. `<space>f  checkov.yaml`
4. `i` and try to edit the file (any other mutation also crashes)

I expected this to happen:
to be able to edit the file

Instead, this happened:
helix crashes:
` 1  [1]    109128 segmentation fault (core dumped)  hx .
    2  framework:                                         %`

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-08-10T09:55:41.443 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T09:56:39.192 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T09:57:50.266 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T09:58:00.933 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T09:58:06.061 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:01:46.123 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:01:54.168 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:02:30.949 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:02:37.030 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:05:19.870 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:07:06.045 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:07:17.146 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:07:33.575 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:08:11.966 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:08:24.717 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:08:31.397 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:10:20.619 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:13:38.568 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:16:16.447 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-10T10:17:48.666 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
```

</details>

### Platform

Linux NixOS 23.05.20230807.011567f (Stoat) x86_64

### Terminal Emulator

alacritty 0.12.2

### Helix Version

helix 23.05 (7f5940be)

**Comment by CptPotato** [comment]
Is it possible for you to try with the latest master?

I'm unable to reproduce the issue, so it might have been fixed in the meantime (by #7621 maybe?).

**Comment by kharf** [comment]
sorry for the issue. 
I couldnt reproduce it with helix 23.05 (57f093d8)

**Comment by pascalkuthe** [comment]
Yeah I think this was almost certainly fixed by #7227
